src/Attributes/Field.java

```diff
@@ -1,50 +1,84 @@
 package Attributes;
 
-import java.util.Objects;
+import com.google.gson.annotations.Expose;
 
-public class Field extends Attribute{
-    private String primitive;
-    private Type type;
-    public Field(String name, String primitive) {
-        super(name);
-        this.primitive = primitive;
-        this.setType();
-    }
-
-
-
-    @Override
-    public Type getType() {
-        return this.type;
-    }
 
+public class Field{
+    @Expose
+    private String primitive;
 
-    public void setType() {
-        this.type = Type.FIELD;
-    }
+    @Expose
+    private String name;
 
-    @Override
-    public String toString() {
-        return (super.toString() + ": " + this.primitive.replaceAll("[\\[\\]]", ""));
+    public Field(String name, String primitive) {
+        if(name == null || name.isEmpty()){
+            throw new IllegalArgumentException("The field name cannot be null or empty");
+        }else if(primitive == null || primitive.isEmpty()){
+            throw new IllegalArgumentException("The field primitive cannot be null or empty");
+        }
+        this.name = name;
+        this.primitive = primitive;
     }
 
-    @Override
+    /**
+     * Description: Gets the name of the primitive type stored in the field.
+     * @return - Returns a string with the type of the primitive.
+     */
     public String getPrimitive() {
         return this.primitive.replaceAll("[\\[\\]]", "");
     }
 
-
-    @Override
+    /**
+     * Description: Sets the primitive type for a field.
+     * @param primitive - The name of the primitive for the field.
+     */
     public void setPrimitive(String primitive) {
+        if(primitive == null || primitive.isEmpty()){
+            throw new IllegalArgumentException("The field primitive cannot be null or empty");
+        }
         this.primitive = primitive;
     }
 
+    /**
+     * Description: Gets the name of the field.
+     * @return - Returns a string with the name of the field.
+     */
+    public String getName() {
+        return name;
```

Comment:
Might want to return this.name.

Reply:
Good catch, I changed it to return this.name